### ISSUE TITLE
Error false positive when arrow function is used as a function argument

### ISSUE THREAD

**Issue by dan-usatoday** [squizlabs/PHP_CodeSniffer#3853]
## Describe the bug
CS throws an error for the code below where test_function accepts a callback function as the second argument. The code runs fine and no PHP errors get thrown. 

### Code sample
```php
test_function( 'foo', fn() => 'bar' );
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
ERROR   | [ ] PHP syntax error: syntax error, unexpected '=>' (T_DOUBLE_ARROW), expecting ')' (Generic.PHP.Syntax.PHPSyntax)
```

## Expected behavior
No error should be thrown.

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | MacOS 11.7
| PHP version | 8.0.21
| PHP_CodeSniffer version | 3.6.0
| Standard | MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend
| Install type | Composer (global/local)

## Please confirm:

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [ ] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@dan-usatoday Could you please check what the _default_ PHP version is on your machine (the one registered in your `PATH`) ? I suspect that is a PHP version below 7.4 (in which arrow functions didn't exist yet).

The `Generic.PHP.Syntax` sniff uses the PHP linter to find parse errors. If you want the sniff to use a different PHP version than the _default_ version on your machine, you can tell PHPCS which PHP to use using `--config-set php_path /path/to/php`.
Ref: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#setting-the-path-to-php

Personally, I'd recommend using a dedicated linter, like [PHP Parallel Lint](https://github.com/php-parallel-lint/PHP-Parallel-Lint/), instead of using PHPCS for finding lint errors.

**Comment by dan-usatoday** [comment]
You're right, it is the PHP version. I'll go ahead and close this. Thanks!
